opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/SpanShimTest.java

```diff
@@ -190,6 +191,25 @@ void log_fields_micros() {
     verifyAttributes(eventData);
   }
 
+  @Test
+  void spanWrapper() {
+    SpanShim shim = new SpanShim(telemetryInfo, span);
+    assertThat(SpanBuilderShim.getSpanShim(shim)).isEqualTo(shim);
+    assertThat(SpanBuilderShim.getSpanShim(new SpanWrapper(shim))).isEqualTo(shim);
+    assertThatThrownBy(() -> SpanBuilderShim.getSpanShim(new SpanWrapper("not a span")))
+        .hasMessage("span wrapper didn't return a span: java.lang.String");
+    assertThatThrownBy(() -> SpanBuilderShim.getSpanShim(null))
+        .hasMessage("span is not a valid SpanShim object: null");
+  }
+
+  @Test
+  void getContextShim() {
+    SpanContextShim contextShim = new SpanContextShim(new SpanShim(telemetryInfo, span));
+    assertThat(SpanBuilderShim.getContextShim(contextShim)).isEqualTo(contextShim);
+    assertThatThrownBy(() -> SpanBuilderShim.getContextShim(null))
+        .hasMessage("context is not a valid SpanContextShim object: null");
+  }
+
```

Comment:
let's add a test to verify that the negative case works, as well (i.e. some other kind of Supplier).

Reply:
the context shim doesn't have a supplier, only the span shim - test case for that is here: https://github.com/open-telemetry/opentelemetry-java/pull/4535/files#diff-f628731aad63f58834336eb865007007c95db889f79cb71adc2e0a0072b297c5R199